force-app/main/default/classes/GlobalGrowth_AccountManagerBatch.cls

```diff
@@ -0,0 +1,71 @@
+/***********************************************************************************************************************        
+Name            : GlobalGrowth_AccountManagerBatch
+Author          : Slalom Team, Reggie Thompson
+Date            : 07/15/24
+@description    : Batch Apex to syncronise Account Owners with Account Managers
+***********************************************************************************************************************/
+public with sharing class GlobalGrowth_AccountManagerBatch implements Database.Batchable<SObject> {
+
+    public String query;
+
+    public GlobalGrowth_AccountManagerBatch(String q) {
+        query = q;
+    }
+
+    /**
+     * @description Start method of the batch
+     * @param bc Batchable Context
+     * @return QueryLocator
+     */
+    public Database.QueryLocator start(Database.BatchableContext bc) {
+        List<Account> accounts = [SELECT Id, OwnerId, Account_Manager__r.pse__Salesforce_User__c, Account_Manager__r.pse__Salesforce_User__r.IsActive FROM Account];
+        Set<Id> accountIds = new Set<Id>();
+
+        for (Account account : accounts) {
+            if (account.OwnerId != account.Account_Manager__r.pse__Salesforce_User__c) {
+                accountIds.add(account.Id);
+            }
+        }
+
+        if (query == null) {
```

Comment:
I think this if statement can/should go at the beginning of the method so that we don't process a soql query and loop over accounts that we're not going to use.

Reply:
Per our call, the query argument (if not null) now overrides the initial SOQL query for Accounts to iterate through instead of the query that's returned and passed to the `execute` method.